tests/Jellyfin.Model.Tests/Dlna/StreamBuilderTests.cs

```diff
@@ -412,7 +439,7 @@ public async Task BuildVideoItemWithDirectPlayExplicitStreams(string deviceName,
                 // Full transcode
                 if (transcodeMode.Equals("Transcode", StringComparison.Ordinal))
                 {
-                    if ((streamInfo.TranscodeReasons & (StreamBuilder.ContainerReasons | TranscodeReason.DirectPlayError)) == 0)
+                    if ((streamInfo.TranscodeReasons & (StreamBuilder.ContainerReasons | TranscodeReason.DirectPlayError | TranscodeReason.VideoRangeTypeNotSupported)) == 0)
```

Comment:
Tbf, this block looks weird.
We are going to transcode. Why can't we use the same codec?

Reply:
It was an old test though, where the transcoding usually happens when input codec is not supported and the transcoding reason video codec not supported is not conclusive enough to determine if it should use another codec. Do we want to change this?

Reply:
I tried:
```cs
if (why.HasFlag(TranscodeReason.VideoCodecNotSupported))
{
    Assert.All(
        videoStreams,
        stream => Assert.DoesNotContain(stream.Codec, streamInfo.VideoCodecs));
}
else
{
    Assert.Contains(targetVideoStream?.Codec, streamInfo.VideoCodecs);
}
```
But `SecondaryAudioNotSupported` and `Full transcode because profile only has ts which does not allow hevc`.
To fix this block, we need to pass the real/full reason or transcoding codec (like `transcodeProtocol`).

Reply:
> To fix this block, we need to pass the real/full reason or transcoding codec (like transcodeProtocol).

We don't have to make it that complicated I think. If the transcode reason that failed the direct play back does make sense we can give a list of reasons that the original codec must be avoided and check those, and don't assume the new codec is same as the original if transcode reason is not any of those due to complex compatibility reasons.

Reply:
Or we can pass `transcodeVideoCodec` and test with it directly.

We already have `transcodeProtocol` (`http`, `HLS.ts`, `HLS.mp4`).

Hmm, we can also introduce a simple struct with protocol, video and audio codecs. This way we can fill it partially.
```cs
new OutputStream { .Protocol = "HLS.mp4", .AudioCodec = "aac" }
```

Maybe left it for the next PR? 🤔

Reply:
Probably yes. Let's limit the scope of this PR to test for the current things, and then move on to use explicit output codecs instead of guessing from transcoding reasons in following PRs.